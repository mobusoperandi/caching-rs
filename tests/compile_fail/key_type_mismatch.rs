use michie::memoized;
use std::collections::HashMap;

#[memoized(key_expr = a, store_type = HashMap<usize, bool>)]
fn f(a: bool) -> bool {
    a
}

fn main() {}
