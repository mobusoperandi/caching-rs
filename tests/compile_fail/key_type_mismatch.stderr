error[E0308]: mismatched types
 --> tests/compile_fail/key_type_mismatch.rs:4:23
  |
4 | #[memoized(key_expr = a, store_type = HashMap<usize, bool>)]
  | ----------------------^-------------------------------------
  | |                     |
  | |                     expected `&usize`, found `&bool`
  | arguments to this function are incorrect
  |
  = note: expected reference `&usize`
             found reference `&bool`
note: function defined here
 --> tests/compile_fail/key_type_mismatch.rs:4:1
  |
4 | #[memoized(key_expr = a, store_type = HashMap<usize, bool>)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/compile_fail/key_type_mismatch.rs:4:23
  |
4 | #[memoized(key_expr = a, store_type = HashMap<usize, bool>)]
  | ----------------------^-------------------------------------
  | |                     |
  | |                     expected `&usize`, found `&bool`
  | arguments to this function are incorrect
  |
  = note: expected reference `&usize`
             found reference `&bool`
note: method defined here
 --> src/lib.rs
  |
  |     fn get(&self, input: &I) -> Option<R>;
  |        ^^^
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
 --> tests/compile_fail/key_type_mismatch.rs:4:23
  |
4 | #[memoized(key_expr = a, store_type = HashMap<usize, bool>)]
  | ----------------------^-------------------------------------
  | |                     |
  | |                     expected `usize`, found `bool`
  | arguments to this function are incorrect
  |
note: method defined here
 --> src/lib.rs
  |
  |     fn insert(&mut self, input: I, return_value: R) -> R;
  |        ^^^^^^
  = note: this error originates in the attribute macro `memoized` (in Nightly builds, run with -Z macro-backtrace for more info)
